packages/shorebird_code_push_client/lib/src/code_push_client.dart

```diff
@@ -67,7 +68,10 @@ class _CodePushHttpClient extends http.BaseClient {
 
   @override
   Future<http.StreamedResponse> send(http.BaseRequest request) {
-    return _client.send(request..headers.addAll(CodePushClient.headers));
+    return retry(
+      () => _client.send(request..headers.addAll(CodePushClient.headers)),
+      retryIf: (e) => e is! CodePushException,
```

Comment:
We should add a test for this. Also I'm not sure this is exactly what we want.

```suggestion
      retryIf: (e) => e is TimeoutException ||  isHttpIOException(e),
```

```dart
bool isHttpIOException(Object e) {
  return e is HttpException ||
      e is TlsException ||
      e is SocketException ||
      e is WebSocketException;
}
```

Reply:
Yep, this is still a draft :)

Reply:
Also instead of baking this into the code_push_client we could just pass a `RetryClient` as the `HttpClient` from `shorebird_cli`:

```dart
RetryClient(
  httpClient,
  when: (response) => response.statusCode >= 500,
  whenError: (e, _) => isHttpIOException(e),
);
```

Reply:
Sounds good